src/miniscript/astelem.rs

```diff
@@ -32,9 +32,28 @@ use script_num_size;
 use ToPublicKey;
 use ToPublicKeyHash;
 use miniscript::types::{self, Property};
+use Miniscript;
+
+impl<Pk, Pkh> AstElem<Pk, Pkh> {
+    /// Internal helper function for displaying wrapper types; returns
+    /// a character to display before the `:` as well as a reference
+    /// to the wrapped type to allow easy recursion
+    fn wrap_char(&self) -> Option<(char, &Box<Miniscript<Pk, Pkh>>)> {
+        match *self {
+            AstElem::Alt(ref sub) => Some(('a', sub)),
+            AstElem::Swap(ref sub) => Some(('s', sub)),
+            AstElem::Check(ref sub) => Some(('c', sub)),
+            AstElem::DupIf(ref sub) => Some(('d', sub)),
+            AstElem::Verify(ref sub) => Some(('v', sub)),
+            AstElem::NonZero(ref sub) => Some(('j', sub)),
+            AstElem::ZeroNotEqual(ref sub) => Some(('u', sub)),
+            _ => None,
+        }
+    }
+}
 
 /// All AST elements
-#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
Why drop `Hash`?

Reply:
I guess I forgot to implement Hash to Miniscirpt type which showed a compile error here. 
And maybe I did this as a quickfix 🤦‍♂ . Will fix it